roadmap/roadmap_2024.md

```diff
@@ -0,0 +1,105 @@
+# Kubebuilder Project Roadmap 2024
+
+### **(Major Release for Kubebuilder CLI 4.x)** Removing Deprecated Plugins for Enhanced Maintainability and User Experience
+
+**Status:** :construction: Work in Progress
+
+**Objective:** To remove all deprecated plugins from Kubebuilder to improve project maintainability and
+enhance user experience. This initiative also includes updating the project documentation to provide clear
+and concise information, eliminating any confusion for users.
+
+**Motivation:** By focusing on removing deprecated plugins—specifically, versions or kinds that can no
+longer be supported—we aim to streamline the development process and ensure a higher quality user experience.
+Clear and updated documentation will further assist in making development workflows more efficient and less prone to errors.
+
+**More Info:** [GitHub Discussion #3622](https://github.com/kubernetes-sigs/kubebuilder/discussions/3622)
+
```

Comment:
nit: we can add `---` to separate each project for better readability.

Reply:
Done